src/clocksi_interactive_coord.erl

```diff
@@ -744,6 +846,22 @@ perform_read({Key, Type}, UpdatedPartitions, Transaction, Sender) ->
             {error, Reason}
     end.
 
+perform_validate_or_read({Key, Type, Token}, UpdatedPartitions, Transaction, Sender) ->
+    Partition = log_utilities:get_key_partition(Key),
+
+    WriteSet = case lists:keyfind(Partition, 1, UpdatedPartitions) of
+        false -> [];
+        {Partition, WS} ->
+            ?INVALID_OBJECT_TOKEN = Token,
```

Comment:
Can this check fail? Seems dangerous to encode an assert like this.

Reply:
I don't expect this check to fail.

If it fails it means that we might return something incorrect and that there is a programming error on the direct caller.